packages/peregrine/src/hooks/__tests__/useWindowSize.spec.js

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import { act } from 'react-test-renderer';
+
+import { useWindowSize } from '../useWindowSize';
+import createTestInstance from '../../util/createTestInstance';
+
+const spies = new Map();
+
+spies.set(
+    'addEventListener',
+    jest.spyOn(window, 'addEventListener').mockImplementation(jest.fn())
+);
+spies.set(
+    'removeEventListener',
+    jest.spyOn(window, 'removeEventListener').mockImplementation(jest.fn())
+);
+
+const Component = () => {
+    useWindowSize();
+
+    return <i />;
+};
+
+test('adds an event listener to the window on mount', () => {
+    createTestInstance(<Component />);
+
+    const spy = spies.get('addEventListener');
+    expect(spy).toHaveBeenCalledTimes(5);
```

Comment:
These numbers don't make sense to me. Even when isolated to just this test it still is called 5 times. Not sure why. Help?

Reply:
I think React is doing something here though I can't figure out what. The 4 calls before the 5th are `error` listeners.